algo-builder-tester/test/v5escrow-test.ts

```diff
@@ -0,0 +1,383 @@
+    const ART = runtime.deployASA(ASSET_NAME, {
+      creator:
+      {
+        ...admin.account,
+        name: "$ART$-creator"
+      }
+    })
+    const aUSD = runtime.deployASA(STABLE_NAME, {
+      creator:
+      {
+        ...admin.account,
+        name: "aUSD-creator"
+      }
+    })
+    artID = ART.assetID
+    usdID = aUSD.assetID
+
+    // opt-in to the asa with alice and billy
+    // runtime.optIntoASA(artID, admin.address, {});
+    // runtime.optIntoASA(usdID, admin.address, {});
+    // creator opts in automatically .
+    runtime.optIntoASA(artID, alice.address, {});
+    runtime.optIntoASA(usdID, alice.address, {});
+    runtime.optIntoASA(artID, billy.address, {});
+    runtime.optIntoASA(usdID, billy.address, {});
+
+    // TODO:ref: split to another function that calls this one.
+    // dispense 1k $ART$ to each alice and billy
+    syncAccounts();
+    const dispenseTxParams: typesW.AssetTransferParam = {
+      type: typesW.TransactionType.TransferAsset,
+      sign: typesW.SignType.SecretKey,
+      fromAccount: admin.account,
+      assetID: artID,
+      payFlags: { totalFee: fee },
+      toAccountAddr: admin.address,
+      amount: 1000,
+    }
+    dispenseTxParams.toAccountAddr = alice.address;
+    runtime.executeTx(dispenseTxParams);
+    dispenseTxParams.toAccountAddr = billy.address;
+    runtime.executeTx(dispenseTxParams);
+  }
+
+  describe("related ASA", function () {
+    it("asset creation and dispense", function () {
+      const adminAssets = admin.createdAssets;
+      syncAccounts();
+      assert.isTrue(adminAssets.has(artID) && adminAssets.has(usdID));
+      // dispensed 1k $ART$ to alice and billy 
+      assert.equal(1000n, alice.assets.get(artID)!['amount']!);
+      assert.equal(1000n, billy.assets.get(artID)!['amount']!);
+      // from admin was $ART$ dispensed
+      assert.equal(999999998000n, admin.assets.get(artID)!['amount']!);
+    })
+
+    it("Asset transfer", function () {
+      syncAccounts();
+      assert.equal(1000n, alice.assets.get(artID)!['amount']!);
+      assert.equal(1000n, billy.assets.get(artID)!['amount']!);
+      const artPayTxParams: typesW.AssetTransferParam = {
+        type: typesW.TransactionType.TransferAsset,
+        sign: typesW.SignType.SecretKey,
+        assetID: artID, // passing asa name is also supported
+        amount: 100,
+        fromAccount: alice.account,
+        toAccountAddr: billy.address,
+        payFlags: { totalFee: fee },
+      }
+      const receipt = runtime.executeTx(artPayTxParams) as typesRT.TxReceipt;
+      console.log('receipt : ', receipt); // DEV_LOG_TO_REMOVE
+      console.log('receipt.txn.snd : ', receipt.txn.snd); // DEV_LOG_TO_REMOVE
+      assert.equal(buf2Hex(receipt.txn.snd), u8a2Hex(parsing.addressToPk(alice.address)))
+      assert.equal(buf2Hex(receipt.txn.arcv!), u8a2Hex(parsing.addressToPk(billy.address)))
+      assert.equal(ab2Hex(receipt.txn.snd.buffer.slice(0, 32)), buf2Hex(receipt.txn.arcv!)) // Why??
```

Comment:
another way to compare buffer/uint8array are https://www.chaijs.com/plugins/chai-bytes/, but buf2Hex etc. looks good too

Reply:
OK, I'll add a note for refactor.

Reply:
resolved: https://github.com/Artcoin-Network/artificial-algorand-contract/pull/6/commits/97f2f5243bc3ae5d17d33ab46f66f9ad30e3db37